pandas/tests/reshape/test_concat.py

```diff
@@ -1279,6 +1279,18 @@ def test_concat_ignore_index(self, sort):
 
         tm.assert_frame_equal(v1, expected)
 
+    def test_concat_same_index_names(self):
+        # GH13475
+        index1 = pd.Index(["a", "b"], name="idx")
+        index2 = pd.Index(["b", "c"], name="idx")
+
+        frame1 = pd.DataFrame({"x": [1, 2]}, index=index1)
+        frame2 = pd.DataFrame({"x": [2, 3]}, index=index2)
+
+        result = pd.concat([frame1, frame2], axis=1)
+
+        assert result.index.name == "idx"
```

Comment:
Here you want to hard code the expected result and check equality with `tm.assert_frame_equal`

Reply:
ah indeed, that's much better. just pushed a remedy.